File: deploy_render.js
const https = require('https');
const API_KEY = "rnd_7tTZNbinUpX85POpKcKRt0HBwvRd";

const options = {
  hostname: 'api.render.com',
  port: 443,
  headers: {
    'Authorization': `Bearer ${API_KEY}`,
    'Accept': 'application/json',
    'Content-Type': 'application/json'
  }
};

function request(path, method = 'GET', data = null) {
  return new Promise((resolve, reject) => {
    const req = https.request({ ...options, path, method }, (res) => {
      let body = '';
      res.on('data', chunk => body += chunk);
      res.on('end', () => {
        if (res.statusCode >= 200 && res.statusCode < 300) {
          try { resolve(JSON.parse(body)); } catch (e) { resolve(body); }
        } else {
          reject(new Error(`API Error ${res.statusCode}: ${body}`));
        }
      });
    });
    req.on('error', reject);
    if (data) req.write(JSON.stringify(data));
    req.end();
  });
}

async function main() {
  try {
    const serviceId = "srv-d9d1euernols73clh690";
    
    // Set Env Vars
    console.log("Setting env vars for service...");
    const envVars = [
      { key: "ENVIRONMENT", value: "production" },
      { key: "DATABASE_URL", value: "postgresql+psycopg://postgres.eohziaiwuuyxxpkhdhoz:XG3zg%2F5*n58Um%2C5@aws-1-ap-northeast-2.pooler.supabase.com:6543/postgres" }
    ];
    const envResult = await request(`/v1/services/${serviceId}/env-vars`, 'PUT', envVars);
    console.log("Success! Updated Env Vars");
    
    // Trigger deploy
    console.log("Triggering deploy...");
    const deployResult = await request(`/v1/services/${serviceId}/deploys`, 'POST');
    console.log("Deploy Triggered:", JSON.stringify(deployResult, null, 2));
  } catch (err) {
    console.error(err.message);
  }
}

main();

File: deploy_render.py
import os
import requests
import json

API_KEY = "rnd_7tTZNbinUpX85POpKcKRt0HBwvRd"
HEADERS = {
    "Authorization": f"Bearer {API_KEY}",
    "Accept": "application/json",
    "Content-Type": "application/json"
}

def main():
    print("Fetching owner ID...")
    res = requests.get("https://api.render.com/v1/owners", headers=HEADERS)
    if not res.ok:
        print("Error fetching owners:", res.text)
        return
    owners = res.json()
    if not owners:
        print("No owners found for this API key.")
        return
    owner_id = owners[0]['owner']['id']
    print(f"Using Owner ID: {owner_id}")

    payload = {
        "type": "web_service",
        "name": "tn-flood-ai-backend",
        "ownerId": owner_id,
        "repo": "https://github.com/Charanroyal418/ai-flood-prediction",
        "autoDeploy": "yes",
        "branch": "main",
        "serviceDetails": {
            "env": "docker",
            "envSpecificDetails": {
                "dockerfilePath": "./backend/Dockerfile"
            },
            "plan": "free",
            "region": "oregon",
            "envVars": [
                {
                    "key": "DATABASE_URL",
                    "value": "sqlite:///./flood_ai.db"
                },
                {
                    "key": "ENVIRONMENT",
                    "value": "production"
                }
            ]
        }
    }

    print("Creating service...")
    res = requests.post("https://api.render.com/v1/services", headers=HEADERS, json=payload)
    if res.ok:
        data = res.json()
        print("Success!")
        print(f"Service URL: {data.get('service', {}).get('serviceDetails', {}).get('url', 'N/A')}")
        print(f"Dashboard URL: {data.get('dashboardUrl', data.get('service', {}).get('dashboardUrl', 'N/A'))}")
    else:
        print("Error creating service:", res.text)

if __name__ == "__main__":
    main()
